x-pack/plugins/aiops/server/routes/categorization_field_validation/define_route.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IRouter } from '@kbn/core/server';
+import type { DataRequestHandlerContext } from '@kbn/data-plugin/server';
+import type { UsageCounter } from '@kbn/usage-collection-plugin/server';
+import { categorizationFieldValidationSchema } from '../../../common/api/log_categorization/schema';
+import { AIOPS_API_ENDPOINT } from '../../../common/api';
+import type { AiopsLicense } from '../../types';
+import { routeHandlerFactory } from './route_handler_factory';
+
+export const defineRoute = (
+  router: IRouter<DataRequestHandlerContext>,
+  license: AiopsLicense,
+  usageCounter?: UsageCounter
+) => {
+  router.versioned
+    .post({
+      path: AIOPS_API_ENDPOINT.CATEGORIZATION_FIELD_VALIDATION,
+      access: 'internal',
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: {
+            body: categorizationFieldValidationSchema,
+          },
+        },
+      },
+      routeHandlerFactory(license, usageCounter)
```

Comment:
I wonder if it would make it easier for subsequent versions if some of the code in these handler functions lived here instead. Like the error catching and extraction of `context`, `request` and `response`.

I'm thinking about what changes we'd need to do to support `v1` when `v2` comes along. We'd probably have to change some of the `request` properties and possibly change the `response` and that code should probably live in this `addVersion` function.

Otherwise we'll have to create a `routeHandlerFactorV1` function which wraps `routeHandlerFactor` and does that manipulation.
Or perhaps that way is fine?

Reply:
Maybe it's best to leave as is as a first step. Since there's a follow up planned anyway which adds a new version for `log_rate_analysis` we could then find out which parts could be moved to this file.

Reply:
Ok yeah, that sounds like a good approach.